error.go

```diff
@@ -156,6 +157,30 @@ func (e *Error) Cause() error {
 	return e.cause
 }
 
+// Is implements the helper interface for errors.Is
+// It first checks if target is an *Error and uses IsOfType to check if e Is target.
+// If target is not an *Error, it instead iterates through the error chain until it reaches a non-*Error error (or nil)
+// and calls errors.Is on that error and target.
+func (e *Error) Is(target error) bool {
+	// Check if target is an *Error, if so we can use IsOfType.
+	t := Cast(target)
+	if t != nil {
+		return e.IsOfType(t.Type())
+	}
+
+	// target is not an *Error, so iterate through the error chain until nil or a non-*Error error is reached and use
+	// errors.Is to check against that.
+	cause := e
+	for cause != nil {
```

Comment:
These is a possible issue here, not unlike the one already discussed about the `Unwrap()` method. 
Opaque wrap, both of errorx and non-errorx error, is supposed to hide the original error. Current impl, on the other hand, allows to 'see' such errors in any wrap scenario.
On the other hand, if we forbid it, then no error from go stdlib, for example, would be inspectable by `errors.Is` once it is wrapped by errorx - even if it is done solely for the purpose of adding a stacktrace, for example.

**Edit**: I checked again and it seems that transparent wrap flag is preserved even if non-errorx error is wrapped. This must be tested, of course, but if that's the case, opaque wrap could legally hide any underlying error, as intended.